### PR TITLE
GPU virtualization with API-forwarding in Kata-containers

### PR DESCRIPTION
**Pull request by brianwang12**
These series of patches introduce Virtio GPU support, leveraging VIRTIO-GPU technology.

1. Enable virtio-gpu for gpu virtualization support
The purpose of this patch is to add virtio-gpu devices to the QEMU command line. This enhancement enables hypervisor to provide GPU virtualization capabilities. which allow the guest OS to leverage the host GPU's capabilities for accelerated 3D rendering.

2.   GPU: Add a configuration fragment
Add kernel configurations related to the virtio-gpu driver to enable GPU virtualization support for containers running with the guest.

3.    Add the virgl option into the qemu configuration
Add additional configurations to enable GPU virtualization support, using virtio-gpu with the qemu hypervisor.

There is simple demo to illustrate 

```
### console from the host machine ###
sudo nerdctl run -p "$VNC_PORT":"$VNC_PORT"  --runtime io.containerd.kata.v2 --env "VNC_PORT=$VNC_PORT" --env "VNC_PW=2022" --env "USER=briwan01"    --device /dev/dri/card0:/dev/dri/card0  --device /dev/dri/renderD128:/dev/dri/renderD128  --rm -it docker.io/consol/ubuntu-xfce1:latest bash
```

```
#### console from the container ####
root@5918d9fc97b3:~# glxinfo -B
name of display: :0
display: :0  screen: 0
direct rendering: Yes
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: Mesa (0x1af4)
    Device: virgl (NV166) (0x1010)
    Version: 23.0.4
    Accelerated: yes
    Video memory: 0MB
    Unified memory: no
    Preferred profile: core (0x1)
    Max core profile version: 4.2
    Max compat profile version: 4.2
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.2
OpenGL vendor string: Mesa
OpenGL renderer string: virgl (NV166)
OpenGL core profile version string: 4.2 (Core Profile) Mesa 23.0.4-0ubuntu1~22.04.1
OpenGL core profile shading language version string: 4.20
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile

OpenGL version string: 4.2 (Compatibility Profile) Mesa 23.0.4-0ubuntu1~22.04.1
OpenGL shading language version string: 4.20
OpenGL context flags: (none)
OpenGL profile mask: compatibility profile

OpenGL ES profile version string: OpenGL ES 3.2 Mesa 23.0.4-0ubuntu1~22.04.1
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.20
```

Fix: https://github.com/kata-containers/kata-containers/issues/7182

### CODE REVIEW

File: src/runtime/pkg/katautils/config.go [hunk 1]
```diff
@@ -93,6 +93,8 @@ type hypervisor struct {
 	MachineType                    string          `toml:"machine_type"`
 	BlockDeviceDriver              string          `toml:"block_device_driver"`
 	EntropySource                  string          `toml:"entropy_source"`
+	VirtioGPU                      string          `toml:"virtio_gpu"`
+	Display                        string           `toml:"display"`
```
Comment (zvonkok) [inline src/runtime/pkg/katautils/config.go:97]: MIssalignment
```suggestion
	Display                        string          `toml:"display"`
```

File: tools/packaging/kernel/build-kernel.sh [hunk 2]
```diff
@@ -302,7 +302,7 @@ get_kernel_frag_path() {
 	# Do not care about options that are in whitelist
 	results=$(grep -v -f ${default_config_whitelist} <<< "$results")
 
-	[[ "${skip_config_checks}" == "true" ]] && echo "${config_path}" && return
+	[[ "${skip_config_checks}" == "true" ]] && echo "{config_path}" && return
```
Comment (zvonkok) [inline tools/packaging/kernel/build-kernel.sh:305]:
```suggestion
	[[ "${skip_config_checks}" == "true" ]] && echo "${config_path}" && return
```

**Comment by jongwu** [comment]
Hi @brianwang12 -, pls format your code and push again.

**Pull request closed without merging**